gallery-so/gallery
# [GAL-4124] Upsell banner for non-wallet user

### Summary of Changes

-  add a wallet banner if the user is not connect their wallet yet.
- Refactoring on how to we handle safe area padding on every screen to adapt the banner
- Added `useManageWalletActions` hooks to trigger the bottom sheet in any screen


### Demo or Before/After Pics



**Banner on every screen + close it**

https://github.com/gallery-so/gallery/assets/4480258/25fd4d61-0d0b-49f9-9694-457c3e559c58


**Connect wallet on banner**


https://github.com/gallery-so/gallery/assets/4480258/5d74ac37-1fe9-4a1a-97da-c19bf884550a

**Restrict in community screen**


https://github.com/gallery-so/gallery/assets/4480258/368dca1c-5dfa-406a-aba6-863cfb30456b





### Edge Cases

1. Press the post button in the bottom navbar
2. Post from community screen
3. Connect wallet from the banner


### Testing Steps

1. Logged in with user that doesnt have any wallet connected
2. You should see a banner at the top
3. You're able to connect the wallet through the banner or close it. 


### Checklist

Please make sure to review and check all of the following:
